File: sdk/python/src/inferlet/runtime.py
"""
Runtime — ``pie:core/runtime``.

Runtime introspection: version, instance ID, available models.
"""

from __future__ import annotations

from wit_world.imports import runtime as _runtime


def version() -> str:
    """Get the runtime version string."""
    return _runtime.version()


def instance_id() -> str:
    """Get the unique instance identifier."""
    return _runtime.instance_id()


def username() -> str:
    """Get the current user's name."""
    return _runtime.username()


def models() -> list[str]:
    """Get names of all available models."""
    return list(_runtime.models())

File: sdk/javascript/src/runtime.ts
// Runtime functions — wraps pie:core/runtime WIT interface.

import * as _rt from 'pie:core/runtime';

/** Returns the runtime version string. */
export function version(): string {
  return _rt.version();
}

/** Returns a unique identifier for the running instance. */
export function instanceId(): string {
  return _rt.instanceId();
}

/** Returns the username of the user who invoked the inferlet. */
export function username(): string {
  return _rt.username();
}

/** Returns a list of all available model names. */
export function models(): string[] {
  return _rt.models();
}
